### ISSUE TITLE
Using a Remote POA network send transactions hang forever and remix transactions are pending forever.

### ISSUE THREAD

**Issue by nickjuntilla** [MetaMask/metamask-extension#4760]
I am using Chrome with Metamask 4.8.0. When connected to a remote POA network via http I can see account balances, but cannot create new transactions. New transactions just hang on the loading screen. Here is an example of a call made through remix from the console that just hangs forever:

`{"jsonrpc":"2.0","id":3644419801,"result":{"difficulty":"0x2","extraData":"0xd88301080b846765746888676f312e31302e31856c696e75780000000000000014fa0cced2a7e2a2d28211f9f64026a5d822bc037e585c5e747b8ce01c33ff7e0aede5466f76ba77a268fa0bf539dc7c444e5f21ce3f4139f9a59b9e96c1547101","gasLimit":"0x989680","gasUsed":"0x0","hash":"0x4870b0001947697e3ab02152f5d095fce4179c760c0d6c43ab1217016bda7666","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","miner":"0x0000000000000000000000000000000000000000","mixHash":"0x0000000000000000000000000000000000000000000000000000000000000000","nonce":"0x0000000000000000","number":"0x3df90","parentHash":"0x1561c8cd921b0e1c99d02df47d62240772325ef3226cf7275dc1d2c5f5b78e27","receiptsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","size":"0x261","stateRoot":"0xd9d3a4e9104f82be4b5dc61f53d984b91eceec9e6bb2f2e1c402308913ccc163","timestamp":"0x5b43c529","totalDifficulty":"0x7bf21","transactions":[],"transactionsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","uncles":[]}}`

This call will be made over and over in the console with a different id.

Here is another call from the console that happens periodically:
{"jsonrpc":"2.0","id":3644419959,"result":"777"}

Here is another example of a call from a send: 
`{"jsonrpc":"2.0","id":4941548250246943,"result":{"difficulty":"0x2","extraData":"0xd88301080b846765746888676f312e31302e31856c696e757800000000000000499475db4418c5722b51049f00191cbcd93c33566ddcea634eac539305adb5f4056a3f7664467dda2ebd6d9bafeb6002839883922b9b05eb5654320d6069d31901","gasLimit":"0x989680","gasUsed":"0x0","hash":"0xf7e557a497bf5387ba20ee56e7fe9e4b7a337d3d9eebe8f3e6448c7188540fb8","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","miner":"0x0000000000000000000000000000000000000000","mixHash":"0x0000000000000000000000000000000000000000000000000000000000000000","nonce":"0x0000000000000000","number":"0x3e23a","parentHash":"0xf32e0de03660b7d2a55604ea8a2f8c7708d3133be7dc82148a0bf138ea59aa45","receiptsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","size":"0x261","stateRoot":"0x497d9103aa8d1dc620786e4fce81d01c55b5e711ae787221fef5f9003d07a730","timestamp":"0x5b301692","totalDifficulty":"0x7c475","transactions":[],"transactionsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","uncles":[]}}`

When I first open metamask there will be like 100 of these kinds of calls. The transactions say complete in metamask, but they do not reflect that in the balance or in remix.

My networkid and chainid are both 777. I have closed my browser and re-installed metamask. The blockchain I am using responds to a remote geth attach from the console and I can send transactions there. I can also send and use transactions on this blockchain from Mist. Something is specifically wrong with metamask. 

*more info eip155Block is 3 on my genesis configuration so that is active. 
I've seen something about eip155Transition in https://github.com/MetaMask/metamask-extension/issues/1260 , but puppeth doesn't create that property only 'eip155Block' so I'm hesitant to add an 'eip155Transition' property to my genesis block. I'm not sure what the value would be if I did.

**Comment by toffoo** [comment]
I'm using Chrome with Metamask 4.8.0 on Mac to use Augur.

If I use the Infura standard Main connection, my transactions broadcast and confirm normally.

If I use my local geth -syncmode light node, my transactions stay "Approved" in Metamask, do not get broadcast, do not confirm, and I get:

"No trusted canonical hash trie" console error.

**Comment by juharris** [comment]
I also still get the spinning issue in 4.16.0. It sometimes happens before I send a transaction that would be rejected.
Here is my solution to get MetaMask back in a usable state:
If MetaMask is spinning and non-responsive:
1. Disable the extension here: chrome://extensions
2. Re-enable the extension.
3. Open the extension.
4. Change the network to use. E.g. Select Main Ethereum Network.
5. Log in to MetaMask.
6. Change the network back to the custom one.
7. Reject any queued transactions.

**Comment by donnfelker** [comment]
I'm having this exact same issue. I was deploying from Remix into Rinkeby. 

@juharris' steps helped me get back into a working state. However, I had two queued transactions. I rejected one, and I got the spinner again. I repeated the steps above, and I was able to come back to Metamask. The transaction I rejected was gone 🎉 , and the other ones still was there. This time, I was able to confirm this transaction and it was deployed on rinkeby.
